bluebox/wave.py

```diff
@@ -0,0 +1,88 @@
+"""wave.py
+
+This file contains the Wave class for generating waveforms.
+It is completely generic and backend agnostic, but does not
+provide any functionality for playing the waveforms. That is
+handled in the backend specific files.
+"""
+
+import typing as t
+import math
+from array import array
+
+
+class SineWave:
+    """SineWave class for generating waveform arrays."""
+
+    _sample_rate: float = 44100.0
+    _channels: int = 1
+
+    def __init__(
+                self,
+                sr: t.Optional[float] = None,
+                ch: t.Optional[int] = None) -> None:
+        """Initialize the Wave object."""
+        if sr is not None:
+            self._sample_rate = sr
+        if ch is not None:
+            self._channels = ch
+
+    def sine(
+            self,
+            freq: float,
+            length: float,
+            amplitude: float = 1.0,
+            phase: float = 0.0) -> t.MutableSequence[float]:
+        """Generate a sine wave.
+
+        Args:
+            freq: The frequency of the sine wave.
+            length: The length of the sine wave in seconds.
+            amplitude: The amplitude of the sine wave.
+            phase: The phase of the sine wave.
+
+
+        Returns:
+            An array representing the sine wave.
+        """
+        wave = array('f')
+
+        # silence / pauses
+        if freq == 0.0 or amplitude == 0.0:
+            for i in range(math.ceil(length * self._sample_rate)):
+                wave.append(0.0)
```

Comment:
If this is a generator, (in fact even if it's not) this would be better as calculating the sum of both points (each * amplitude/2, assuming an equal mix)